src/interactors/LearningObjectInteractor.ts

```
@@ -622,58 +803,72 @@ export class LearningObjectInteractor {
+          ResourceErrorReason.INVALID_ACCESS,
         );
-        const objectIds = objectRefs.map(obj => obj.id);
-        // Remove objects from library
-        await params.library.cleanObjectsFromLibraries(objectIds);
-        // Delete objects from datastore
-        await params.dataStore.deleteMultipleLearningObjects(objectIds);
-        // For each object id
-        objectRefs.forEach(async obj => {
-          // Attempt to delete files
-          const path = `${params.user.username}/${obj.id}/`;
-          params.fileManager.deleteAll({ path }).catch(e => {
-            console.error(`Problem deleting files at ${path}. ${e}`);
+      }
+
+      // Get LearningObject ids
+      const objectRefs: {
+        id: string;
+        parentIds: string[];
+      }[] = await Promise.all(
+        learningObjectNames.map(async (name: string) => {
+          const authorId = await this.findAuthorIdByUsername({
+            dataStore,
+            username: user.username,
           });
-          // Update parents' dates
-          updateParentsDate({
-            dataStore: params.dataStore,
-            parentIds: obj.parentIds,
-            childId: obj.id,
-            date: Date.now().toString(),
+          const id = await dataStore.findLearningObject({
+            authorId,
+            name,
           });
+          const parentIds = await dataStore.findParentObjectIds({
+            childId: id,
+          });
+          return { id, parentIds };
+        }),
+      );
+      const objectIds = objectRefs.map(obj => obj.id);
+      // Remove objects from library
+      await library.cleanObjectsFromLibraries(objectIds);
+      // Delete objects from datastore
+      await dataStore.deleteMultipleLearningObjects(objectIds);
+      // For each object id
+      objectRefs.forEach(async obj => {
+        // Attempt to delete files
+        const path = `${user.username}/${obj.id}/`;
+        fileManager.deleteAll({ path }).catch(e => {
```

We should report this error out to sentry as well so we don't have to check logs manually.